Rich-Text-Editor-(RTE)-Overview.md

```diff
@@ -1,117 +1,137 @@
-
-The widget definition in detail is as follows:
-#### button
-Specifies a button label for the widget.
-#### inline
-Specifies whether the widget is an inline component or block component. The link and the math component are inline and the rest are declared block components.
-#### template
-It specifies the wrapper template for the widget.
-#### edit function
-This method will be executed when a widget is being edited. In this function the default action is canceled since we have used our own edit modal. `RteHelperService._openCustomizationModal()`is a helper function for opening the modal which is used to insert new components or editing existing ones. It uses the customizationArgSpecs (obtained from the component definition) to know what the editable fields are for each component, which allows it to render the modal properly for any component.
-#### downcast function
-Downcasting a widget means removing any property which is not needed in the outputs source. We downcast the widget here to render the html content without any wrappers.
-
-The downcast function is used to downcast the widget instance by clearing the angular rendering content and returning the rich text component without any wrapper.
-#### upcast function
-Upcasting an element means changing that element into a widget. Upcast function decides which element will become a widget.
-
-The upcast function is used to upcast an element to this widget. It returns whether an element is an instance of the widget. The element will be upcasted if it is an instance of the widget.
-#### data function
-This function will be executed every time the widget data changes. It will set the attributes of rich text components according to the change in data values.
-#### init function
-This function is executed while initialising a widget, after a widget instance is created, but before it is ready. It is executed before the first time when data function is exceuted. It reads values from component attributes and saves them.
-
-# Hidden components
+`CkEditor4RteComponent` is the Angular component, defined in [`ck-editor-4-rte.component.ts`](https://github.com/oppia/oppia/blob/develop/core/templates/components/ck-editor-helpers/ck-editor-4-rte.component.ts). There we define a `ckConfig` object that configures the editor. This configuration includes the `toolbar` attribute, which specifies the order in which rich text components appear in the editor toolbar.
+
+Near the end of the `ngAfterViewInit()` function, we use `ck.on` to specify the following callback functions:
+
+* The function passed to `ck.on('instanceReady', ...)` is executed when any new instance of CKEditor is ready. For example it will be executed when we click on the content of a card, for each option that we have for multiple choice options etc. Here we can add the css and icons to the buttons in the toolbar.
```

Comment:
Could you please do an edit pass on this and the sections below? Some of the text seems not that grammatical, e.g. lack of parallel structure in "we click on..." and "for each option..."; "css" should be uppercase, not clear what "here" refers to, etc.

Reply:
done